### DISCUSSION TITLE
Does `useAtomDevtools` run in production?

### DISCUSSION THREAD

**Discussion by saadq** [pmndrs/jotai#865] [Q&A] [answered]
Just wanted to get confirmation on this – does using `useAtomDevtools` always run or is there a guard to prevent it running in `production`? Because it's a hook, I'm not sure if the client could handle conditionally using it, so just wanted to make sure.

**Comment by dai-shi** [comment, marked as answer]
The current `useAtomDevtools` implementation doesn't have a guard. So, you would need to do it on your end.

Something like this?
```js
import { useAtomDevtools as useAtomDevtoolsOrig } from 'jotai/devtools'

const useAtomDevtools = process.env.NODE_ENV === 'production' ? () => {} : useAtomDevtoolsOrig
```
